### PR TITLE
Updates for the writing docs section

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request starts addressing some of the feedback necessary for the migration of the documentation towards 6.0. In this particular case the writing docs section.

What was done:
- The links to the documentation are now updated.
- Added more context to the doc-block page to inform the reader how he can add a description to a proptype (the typescript snippet is not added, will be added once i start making the changes for the snippets chore).
- The docs page's snippet referencing List and List item is now updated to match the image.
- Created a new page to show how the reader can preview his/hers documentation and also how to build it (with `start-storybook --docs` and `build-storybook --docs` flag usage).
- The change from `Preview` to `Canvas`,  mentioned [here](https://github.com/storybookjs/storybook/pull/11654#discussion_r459845640) is addressed. I would like your input on this to see if it's ok with all of you.

Feel free to provide feedback.

### CODE REVIEW

File: docs/writing-docs/doc-blocks.md [hunk 1]
```diff
@@ -17,9 +21,39 @@ Storybook Docs automatically generates component props tables for components in
...
+  isDisabled: PropTypes.bool.isRequired,
+  /**
+  The display content of the button
+  */
+  content: PropTypes.string.isRequired,
+};
+
+```
+
+By including the additional information, the args tables will be updated. Offering a richer experience for any stakeholders involved.
```
Comment (tmeasday) [inline docs/writing-docs/doc-blocks.md:51]: I guess it is "args table" (not plural) right?

File: docs/writing-docs/doc-blocks.md [hunk 2]
```diff
@@ -17,9 +21,39 @@ Storybook Docs automatically generates component props tables for components in
...
+
+
 #### DocsPage
 
-To use the ArgsTable in [DocsPage](./docspage#component-parameter), export a component property on your stories metadata:
+To use the ArgsTable in [DocsPage](./docs-page.md#component-parameter), export a component property on your stories metadata:
```
Comment (tmeasday) [inline docs/writing-docs/doc-blocks.md:56]: I think if it is capitalised it should be in code ticks (ie either use "args table" [informal] or `ArgsTable` [referring to the doc block component]). Wdyt @jonniebigodes ?

File: docs/writing-docs/doc-blocks.md [hunk 3]
```diff
@@ -241,7 +275,7 @@ Stories (component examples) are the basic building blocks in Storybook. In Stor
 
 #### DocsPage
 
-In DocsPage, a `Story` block is generated for each story in your [CSF] file, it's wrapped with a `Preview` wrapper that gives it a toolbar on top (in the case of the first “primary” story) and a source code preview underneath.
+In DocsPage, a `Story` block is generated for each story in your [CSF](../api/csf.md) file, it's wrapped with a `Preview` wrapper that gives it a toolbar on top (in the case of the first “primary” story) and a source code preview underneath.
```
Comment (tmeasday) [inline docs/writing-docs/doc-blocks.md:278]: Should this be canvas?

**Review by tmeasday — APPROVED** [review]
Looks good. A few tweaks

File: docs/writing-docs/doc-blocks.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/writing-docs/doc-blocks.md]: You're correct @tmeasday , fixed with the latest commit.

File: docs/writing-docs/doc-blocks.md (hunk 2 above)
Comment (jonniebigodes) [reply docs/writing-docs/doc-blocks.md]: @tmeasday you're absolutely correct. I've adjusted accordingly.

File: docs/writing-docs/doc-blocks.md (hunk 3 above)
Comment (jonniebigodes) [reply docs/writing-docs/doc-blocks.md]: fixed with the latest commit

**Comment by jonniebigodes** [comment]
merging as is, further improvements will be made with future pull requests.

**Pull request merged**